src/api/libcellml/analyserequationast.h

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright libCellML Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include "libcellml/analyser.h"
+#include "libcellml/generator.h"
+
+#undef NAN
+
+namespace libcellml {
+
+/**
+ * @brief The AnalyserEquationAst class.
+ *
+ * The AnalyserEquationAst class is for representing an equation AST in the
+ * context of a CellML Analyser.
+ */
+class LIBCELLML_EXPORT AnalyserEquationAst
+{
+    friend class Analyser;
+    friend class Generator;
+
+public:
+    enum class Type
+    {
+        // Assignment.
```

Comment:
See earlier comment - there's also a "sep" listed in the specifications? Not sure what it does though ...

Reply:
The `<sep>` is used for numbers that follow the e-notation. So, it's handled (see other comment) and does't need to be listed here.